### ISSUE TITLE
Amplify cannot find .dotnet SDK

### ISSUE THREAD

**Issue by CoreyPritchard** [aws-amplify/amplify-cli#4226]
**Describe the bug**
I'm following the [Introducing .NET Core Support for AWS Amplify Backend Functions](https://aws.amazon.com/blogs/developer/introducing-net-core-support-for-aws-amplify-backend-functions/) documentation. I am at the `amplify push` part of the documentation. 

When I run `amplify push`, I get the following error:
```
Unable to detect required dependencies:
- The .NET Core 3.1 SDK must be installed. It can be installed from https://dotnet.microsoft.com/download

× An error occurred when pushing the resources to the cloud

Missing required dependencies to package ServerlessDotnet
An error occured during the push operation: Missing required dependencies to package ServerlessDotnet
```

However, I then run `dotnet --info` and see this:
```
.NET Core SDK (reflecting any global.json):
 Version:   3.1.201
 Commit:    b1768b4ae7

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.201\
```
This indicates to me that I do successfully have .NET Core 3.1 installed, but it can't be found?

**Amplify CLI Version**
4.19.0

**To Reproduce**
Without .dotnet 3.1.201 installed, follow the [documentation](https://aws.amazon.com/blogs/developer/introducing-net-core-support-for-aws-amplify-backend-functions/) up to the `amplify push` part. Run the push command and notice the error. Download .net 3.1.201. Restart powershell. Run `amplify push` again and notice the error is still present.

**Expected behavior**
I expect what the documentation says would happen next:
> The deployment process will compile and package our .NET code for deployment, and after a few minutes, a new Lambda function will be created that will run our code

**Screenshots**
N/A. See the console output above

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Node Version: v10.15.10

**Additional context**
Angular version: 9.1.5

**Comment by jhockett** [comment]
Hi @CoreyPritchard, could you please try to see if using Command Prompt instead of PowerShell yields a different result?

**Comment by CoreyPritchard** [comment]
> Hi @CoreyPritchard, could you please try to see if using Command Prompt instead of PowerShell yields a different result?

I am getting the same result. I also tried Mintty and got the same result.

**Comment by UnleashedMind** [comment]
There might be some issue with the PATH settings on the platform that you run the command. 
 
In the same shell that you run `amplify push`, try to run `which dotnet` and see if it prints out the path to the dotnet executable.

**Comment by CoreyPritchard** [comment]
@UnleashedMind My path is set to `C:\Program Files\dotnet\`, which has the 3.1 sdk in the `sdk` folder...i.e. `C:\Program Files\dotnet\sdk\3.1.201`. Should I set my path specifically to this versioned folder?

**Comment by UnleashedMind** [comment]
that's not neccessary, as long as the system can find the `dotnet` executable. So when you run `which dotnet`, what's the printout?

**Comment by CoreyPritchard** [comment]
Using powershell, `which dotnet` doesn't work:
```
which : The term 'which' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ which dotnet
+ ~~~~~
    + CategoryInfo          : ObjectNotFound: (which:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

However, when I use `dotnet --info` I get this:
```
.NET Core SDK (reflecting any global.json):
 Version:   3.1.201
 Commit:    b1768b4ae7

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.201\

Host (useful for support):
  Version: 3.1.3
  Commit:  4a9f85e9f8

.NET Core SDKs installed:
  1.0.0-preview4-004233 [C:\Program Files\dotnet\sdk]
  1.0.4 [C:\Program Files\dotnet\sdk]
  1.1.12 [C:\Program Files\dotnet\sdk]
  1.1.14 [C:\Program Files\dotnet\sdk]
  2.1.4 [C:\Program Files\dotnet\sdk]
  2.1.104 [C:\Program Files\dotnet\sdk]
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.302 [C:\Program Files\dotnet\sdk]
  2.1.502 [C:\Program Files\dotnet\sdk]
  2.1.504 [C:\Program Files\dotnet\sdk]
  2.1.801 [C:\Program Files\dotnet\sdk]
  3.0.100-preview8-013656 [C:\Program Files\dotnet\sdk]
  3.1.101 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview8.19405.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview8-28405-07 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-preview8-28405-07 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by UnleashedMind** [comment]
try to use the "where.exe" command on Windows

https://stackoverflow.com/questions/304319/is-there-an-equivalent-of-which-on-the-windows-command-line

**Comment by CoreyPritchard** [comment]
Gotcha, just used it. Got this:
```
C:\Program Files\dotnet\dotnet.exe
```

**Comment by UnleashedMind** [comment]
And in the same shell you just successfully ran the `where` command, `cd` to the project's root directory, and try to run `amplify push` again. 
See if you got the same error message or another error message asking you to install some dotnet dependencies.

**Comment by CoreyPritchard** [comment]
I ran it in the project's root directory earlier, but just did it again with the same error: 

![image](https://user-images.githubusercontent.com/36486101/81618500-7a64a000-93b5-11ea-9688-c6cf757d9d19.png)

**Comment by CoreyPritchard** [comment]
Anything needed from me on this? I'd like to get through this demo so I can start moving some production code!

**Comment by edwardfoyle** [comment]
Hi @CoreyPritchard this looks like a bug in our version detection logic when there are multiple versions of the .NET SDK installed. If you don't need the previous versions of the SDK, you can uninstall them and then the check should pass. If you do need them, then unfortunately there's not an easy workaround until we release the fix above. The only way would be to [set up amplify for local development](https://github.com/aws-amplify/amplify-cli#developing), then make the change in  the above PR locally.

**Comment by CoreyPritchard** [comment]
@edwardfoyle Thanks for getting back to me about this!

It's good to know at least what the cause of the issue is. I can temporarily try to find a work around or just use the one SDK for the time being (I've been meaning to upgrade some of our projects to .net 3.1 anyways).

How long would it take for something like this to be released?

**Comment by CoreyPritchard** [comment]
I can confirm that the work around of removing other SDKs does work successfully!

**Comment by edwardfoyle** [comment]
@CoreyPritchard This should be fixed in the latest version of the CLI (4.20.0)
